### PR TITLE
REP-4491 Handling options cadf method and unknown cadf methods

### PR DESCRIPTION
**Pull request by dmnjohns**
This branch is not expected to be merged. This PR is to make code review easy before a release is cut off of this branch.

### CODE REVIEW

**Comment by Mario-Lopez** [comment]
The changes made look good.  I'd love to see a functional test for this including a test for the unknown path (maybe TRACE?).  It would look similar to this one:

https://github.com/rackerlabs/repose/blob/master/repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/herp/HerpCloudfeedCADFdefaultTemplateTest.groovy#L103

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/herp/HerpCloudfeedCADFdefaultTemplateTest.groovy [hunk 1]
```diff
@@ -58,12 +58,14 @@ class HerpCloudfeedCADFdefaultTemplateTest extends ReposeValveTest {
...
-                'DELETE': 'update/delete',
-                'PATCH' : 'update/patch'
+                'GET'    : 'read/get',
+                'HEAD'   : 'read/head',
+                'OPTIONS': 'read/options',
+                'POST'   : 'update/post',
+                'PUT'    : 'update/put',
+                'DELETE' : 'update/delete',
+                'PATCH'  : 'update/patch',
+                'TRACE'  : 'unknown/trace'
```
Comment (Mario-Lopez) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/herp/HerpCloudfeedCADFdefaultTemplateTest.groovy]: No need to change this, but just fyi, the quotations on the keys here are extraneous.  You only need them when there's a character in the key name (e.g. a dash).

Comment (dmnjohns) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/herp/HerpCloudfeedCADFdefaultTemplateTest.groovy]: The more you know. 🌠

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/herp/HerpCloudfeedCADFdefaultTemplateTest.groovy [hunk 2]
```diff
@@ -129,13 +131,17 @@ class HerpCloudfeedCADFdefaultTemplateTest extends ReposeValveTest {
...
-        "415"        | "reposeUser1" | "/resource1/id/dddddddddddd" | "PATCH" | "some data" | "UNSUPPORTED_MEDIA_TYPE"
-        "413"        | "reposeTest"  | "/resource1/id/eeeeeeeeeeee" | "PUT"   | "some data" | "PAYLOAD_TOO_LARGE"
-        "500"        | "reposeTest1" | "/resource1/id/ffffffffffff" | "PUT"   | "some data" | "INTERNAL_SERVER_ERROR"
+        responseCode | username      | request                      | method    | reqBody     | respMsg
+        "404"        | "User"        | "/resource1/id/aaaaaaaaaaaa" | "GET"     | ""          | "NOT_FOUND"
+        "405"        | "testUser"    | "/resource1/id"              | "POST"    | ""          | "METHOD_NOT_ALLOWED"
+        "405"        | "testUser"    | "/resource1/id"              | "HEAD"    | ""          | "METHOD_NOT_ALLOWED"
+        "405"        | "testUser"    | "/resource1/id"              | "OPTIONS" | ""          | "METHOD_NOT_ALLOWED"
+        "405"        | "testUser"    | "/resource1/id"              | "DELETE"  | ""          | "METHOD_NOT_ALLOWED"
+        "405"        | "testUser"    | "/resource1/id"              | "TRACE"   | ""          | "METHOD_NOT_ALLOWED"
```
Comment (Mario-Lopez) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/herp/HerpCloudfeedCADFdefaultTemplateTest.groovy]: 👍

**Comment by Mario-Lopez** [comment]
Changes look good.  I guess I'll take a look again when there's a PR for the desired branch.

**Comment by dmnjohns** [comment]
Oh, this PR should be against the desired branch now. I changed it in-flight. 😛

**Comment by dmnjohns** [comment]
Side note: I kicked off the Jenkins job to test this, but that will take... a while.

**Comment by Mario-Lopez** [comment]
Build is here:  https://jenkins.openrepose.org/job/repose-pull_requests/1299/

**Comment by Mario-Lopez** [comment]
Build was successful.

**Pull request merged**